docs-website/topics/serialization-json-io-sources.md

```diff
@@ -0,0 +1,280 @@
+
+If your input contains multiple JSON objects in a top-level JSON array, you can use [`.decodeToSequence()`](https://kotlinlang.org/api/kotlinx.serialization/kotlinx-serialization-json/kotlinx.serialization.json/decode-to-sequence.html) to process the elements lazily.
+This lets you handle each value as it is parsed.
+
+Here's an example:
+
+```kotlin
+// Imports declarations from the serialization library
+import kotlinx.serialization.*
+import kotlinx.serialization.json.*
+import java.io.FileInputStream
+
+@Serializable
+data class Project(val name: String, val stars: Int)
+
+fun main() {
+    // Opens an InputStream for the projects.json file containing a JSON array of Project objects
+    FileInputStream("projects.json").use { input ->
+
+        // Lazily deserializes each Project from the InputStream
+        val projects = Json.decodeToSequence<Project>(input)
+
+        // Processes elements one by one
+        for (project in projects) {
+            println(project)
+        }
+    }
+}
+```
+
+> You can iterate through sequences returned by `.decodeToSequence()` only once
+> because they are tied to the underlying stream.
+> 
+{style="note"}
+
+## JSON serialization with kotlinx-io and Okio
+
+In addition to JVM streams, you can work with JSON using I/O types, such as `Source` and `Sink` from [`kotlinx-io`](https://github.com/Kotlin/kotlinx-io), and `BufferedSource`
+and `BufferedSink` from [Okio](https://square.github.io/okio/).
+
+You can use the following `Json` extension functions to read and write JSON directly through these I/O types:
+
+* [`.encodeToSink()`](https://kotlinlang.org/api/kotlinx.serialization/kotlinx-serialization-json-io/kotlinx.serialization.json.io/encode-to-sink.html) and [`.encodeToBufferedSink()`](https://kotlinlang.org/api/kotlinx.serialization/kotlinx-serialization-json-okio/kotlinx.serialization.json.okio/encode-to-buffered-sink.html) to write JSON to a `Sink` or `BufferedSink`.
+* [`.decodeFromSource()`](https://kotlinlang.org/api/kotlinx.serialization/kotlinx-serialization-json-io/kotlinx.serialization.json.io/decode-from-source.html) and [`.decodeFromBufferedSource()`](https://kotlinlang.org/api/kotlinx.serialization/kotlinx-serialization-json-okio/kotlinx.serialization.json.okio/decode-from-buffered-source.html) to read a single JSON value from a `Source` or `BufferedSource`.
+* [`.decodeSourceToSequence()`](https://kotlinlang.org/api/kotlinx.serialization/kotlinx-serialization-json-io/kotlinx.serialization.json.io/decode-source-to-sequence.html) and [`.decodeBufferedSourceToSequence()`](https://kotlinlang.org/api/kotlinx.serialization/kotlinx-serialization-json-okio/kotlinx.serialization.json.okio/decode-buffered-source-to-sequence.html) to lazily decode multiple JSON values as a `Sequence<T>`.
+
+> The JSON I/O extension functions use UTF-8 encoding and throw `SerializationException` for invalid JSON data and `IOException` for I/O failures.
```

Comment:
This note is applicable to JVM streams as well. I believe we should move it to the top of the document.